### PR TITLE
util: add an interface for Chunk to count the memory usage

### PR DESCRIPTION
**Pull request by XuHuaiyu**
As the title says,
PTAL @zz-jason @coocood

### CODE REVIEW

File: util/chunk/chunk_test.go [hunk 1]
```diff
@@ -394,6 +394,43 @@ func (s *testChunkSuite) TestGetDecimalDatum(c *check.C) {
 	c.Assert(decDatum.Frac(), check.Equals, decFromChk.Frac())
 }
 
+func (s *testChunkSuite) TestChunkMemoryUsage(c *check.C) {
```
Comment (ngaut) [inline util/chunk/chunk_test.go]: Could you add case for date/time/datatime/blob ?

File: util/chunk/chunk.go [hunk 2]
```diff
@@ -57,6 +57,18 @@ func NewChunkWithCapacity(fields []*types.FieldType, cap int) *Chunk {
...
 
+// MemoryUsage returns the total memory usage of a Chunk in MB.
+// We ignore the size of column.length, column.nullCount and column.elemBuf
+// since they have little effect of the total memory usage.
+func (c *Chunk) MemoryUsage() float64 {
+	sum := int64(0)
+	for _, c := range c.columns {
+		curColMemUsage := int64(cap(c.nullBitmap)) + int64(cap(c.offsets)*4) + int64(cap(c.data)) + int64(cap(c.elemBuf))
+		sum += curColMemUsage
+	}
```
Comment (coocood) [inline util/chunk/chunk.go]: Add the memory of `column` structs?

File: util/chunk/chunk_test.go (hunk 1 above)
Comment (XuHuaiyu) [reply util/chunk/chunk_test.go]: blob has no difference with varchar
I add cases for duration and time

**Comment by XuHuaiyu** [comment]
PTAL @coocood @ngaut

File: util/chunk/chunk.go [hunk 3]
```diff
@@ -57,6 +57,17 @@ func NewChunkWithCapacity(fields []*types.FieldType, cap int) *Chunk {
 	return chk
 }
 
+// MemoryUsage returns the total memory usage of a Chunk in B.
+// We ignore the size of column.length and column.nullCount
+// since they have little effect of the total memory usage.
+func (c *Chunk) MemoryUsage() (sum int64) {
+	for _, c := range c.columns {
```
Comment (coocood) [inline util/chunk/chunk.go]: The local name `c` conflicts with the method receiver `(c *Chunk)`

**Comment by XuHuaiyu** [comment]
PTAL @coocood

**Comment by coocood** [comment]
LGTM

**Pull request merged**